src/alert.cpp

```diff
@@ -245,8 +247,20 @@ bool CAlert::ProcessAlert(const std::vector<unsigned char>& alertKey, bool fThre
     return true;
 }
 
-void
-CAlert::Notify(const std::string& strMessage, bool fThread)
+// static
+void CAlert::NotifyInternal(const std::string& strMessage)
+{
+    static CAlert internalAlert;
+    internalAlert.vchMsg = std::vector<unsigned char>(strMessage.begin(), strMessage.end());
+    uint256 zero;
+    LOCK(cs_mapAlerts);
```

Comment:
It would be better to hold the cs_mapAlerts lock for as little time as possible, so good to add a local scope like this:
```
{
    LOCK(cs_mapAlerts);
    mapAlerts[zero] = internalAlert;
}
```